### DISCUSSION TITLE
k0s Kubernetes

### DISCUSSION THREAD

**Discussion by tteck** [tteck/Proxmox#877] [Announcements]
I've never messed with Kubernetes.  https://k0sproject.io/

Test away...

```
bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/ct/k0s-v5.sh)"
```

config file `/etc/k0s/k0s.yaml`

**Comment by liveware-problems** [comment]
Has anyone tried this at all, from the 4 upvotes and 1 thumbs up?

I created a new LXC using the script, changed from dhcp to static IP and then tried k0s and it wasn't running; update the IP in the `k0s.yaml` and still NG. Deleted the LXC, used Advanced when using the script again to set the static IP, but k0s still isn't working. Removed the services, downloaded k0s again using the steps from https://docs.k0sproject.io/v0.13.0/install/#installation-steps and still NG. Error seems to be around metric-server, with `kubectl get nodes` giving an error and `kubectl get pods` showing 2 pods that aren't starting....



```
root@k0s:/etc/k0s# k0s kubectl get nodes
E0222 17:37:07.119142  294810 memcache.go:255] couldn't get resource list for metrics.k8s.io/v1beta1: the server is currently unable to handle the request
E0222 17:37:07.126257  294810 memcache.go:106] couldn't get resource list for metrics.k8s.io/v1beta1: the server is currently unable to handle the request
E0222 17:37:07.127671  294810 memcache.go:106] couldn't get resource list for metrics.k8s.io/v1beta1: the server is currently unable to handle the request
E0222 17:37:07.129049  294810 memcache.go:106] couldn't get resource list for metrics.k8s.io/v1beta1: the server is currently unable to handle the request
No resources found
root@k0s:/etc/k0s# k0s kubectl get pods --all-namespaces
E0222 17:37:59.596422  294906 memcache.go:255] couldn't get resource list for metrics.k8s.io/v1beta1: the server is currently unable to handle the request
E0222 17:37:59.602826  294906 memcache.go:106] couldn't get resource list for metrics.k8s.io/v1beta1: the server is currently unable to handle the request
E0222 17:37:59.604253  294906 memcache.go:106] couldn't get resource list for metrics.k8s.io/v1beta1: the server is currently unable to handle the request
E0222 17:37:59.605938  294906 memcache.go:106] couldn't get resource list for metrics.k8s.io/v1beta1: the server is currently unable to handle the request
NAMESPACE     NAME                              READY   STATUS    RESTARTS   AGE
kube-system   coredns-9864b985-wq626            0/1     Pending   0          2d1h
kube-system   metrics-server-7446cc488c-6fwhz   0/1     Pending   0          2d1h
```
Any advice / help from someone who has it working would be appreciated!

**Reply by AAH134** [reply]
I tried this and many things to get it to work with no luck,

**Comment by liveware-problems** [comment]
I found the following thread on the Proxmox support forums 

https://forum.proxmox.com/threads/kubernetes-sharing-of-dev-kmsg-with-the-container.61622/

Not sure if this is something that can be handled within the install script? I've not actually tried all the config settings in the post (and the linked Medium post) yet to verify if they actually work and allow k0s to start inside the LXC though - not enough time away from work!

**Comment by JonTheNiceGuy** [comment]
I think this has deployed a single controller node (previously referred to as a master) and no worker nodes. I'll have a play over the next few days and see if I can get it to deploy some workers too.

**Reply by beautyfree** [reply]
any progress?

**Reply by JonTheNiceGuy** [reply]
Apologies - other things came up and I pushed it to one side! I'm not sure when I'll get a chance. Sorry :(
